### ISSUE TITLE
GitHub and Enterprise Cloud / Server connector client (8.9 / 8.10 Beta)

### ISSUE THREAD

**Issue by danajuratoni** [elastic/connectors-python#870]
## New GitHub and Enterprise Cloud / Server connector
New connector client to be released with 8.9 in Beta.
Preliminary date for 8.9 Feature Freeze: Tue, Jun 20 2023

### Acceptance criteria
Support for 
* GitHub Cloud (compatible with [GH Free for organizations](https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-free-for-organizations), [GH Team](https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-team), [GitHub Enterprise Cloud](https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-enterprise))
* GitHub Enterprise Server

===
- [x] TDD in [GDrive](https://drive.google.com/drive/folders/1DN2y4TRXIuybX8vawHljkWozoyPEOYaf)
- [x] TDD reviewed by engineer and PM
- [x] QA plan in [GDrive](https://drive.google.com/drive/folders/1DN2y4TRXIuybX8vawHljkWozoyPEOYaf)
- [x] QA plan reviewed by engineer and PM
- [x] Code merged in main following the defined PR process
     - [x] #915
     - [x] #1094
     - [x] #1100
     - [x] #1288
     - [x] #1295
- [x] https://github.com/elastic/connectors-python/issues/950
- [x] Functional tests performed
- [x] Performance test/ reports against the last stable release branch of the connector framework and in Elastic Cloud 
- [x] Connector client documentation merged in repo 
     - [x] #1118
- [x] Demo of the connector in latest builds in Elastic Cloud in [GDrive](https://drive.google.com/drive/folders/1DN2y4TRXIuybX8vawHljkWozoyPEOYaf)
- [x] Sign off given from an engineer in Ingestion
     - [x] Code backported to the correct version 
     - [x] Automated tests to 90% coverage
- [x] Regression test suite performed for the release along with QA day for all supported OS in https://www.elastic.co/support/matrix 
- [x] Compatibility tests for running connector as native in Cloud

**Comment by danajuratoni** [comment]
@timgrein updated the AC with docs "reviewed by engineer and PM", marking the TDD as reviewed based on your comment in chat.
@prashant-elastic please assign an engineer to the epic. Thanks!

**Comment by abhishekjoshi-crest** [comment]
**Updates [4th May - 10th May]**

 - Addressed Comments in GitHub TDD
 - Started working on Ping Functionality
 - Started writing test cases

**Comment by abhishekjoshi-crest** [comment]
Updates [11th May - 17th May]

- Completed the implementation and testing of ping functionality
- Raised a [PR](https://github.com/elastic/connectors-python/pull/917) for Ping Feature
- Currently, addressing PR comments

**Comment by abhishekjoshi-crest** [comment]
**Updates - [18th May - 24th May]**



- Ping PR is merged
- Start implementation of indexing and it is in progress

**Comment by abhishekjoshi-crest** [comment]
**Updates [25th May - 31st May]**

- Implementation of indexing is on going

**Comment by abhishekjoshi-crest** [comment]
**Updates [1st June - 7th June]**

- Internal testing is ongoing for GitHub indexing.
- Trying different ways to optimize the result

**Comment by abhishekjoshi-crest** [comment]
**Updates [15th June - 21st June]**
- Internal testing was completed
- Raised PR for indexing and addressing comments of Tim
- Raised PR for E2E Test

**Comment by abhishekjoshi-crest** [comment]
**Updates [22th June - 28th June]**
- All the Functional PRs are merged into connector branch.
    - #1095 
    - #1101 
- Connector moved to main.
    - #1130 
- Documentation review is ongoing.
    - #1118 
- Started working on GraphQL implementation.

**Comment by abhishekjoshi-crest** [comment]
Updates [29th June - 5th July]

- Working on GraphQL Implementation for GitHub.

**Comment by abhishekjoshi-crest** [comment]
**Updates [6th July - 12th July]**
- Internal testing is ongoing
- Currently working on E2E implementation as per GraphQL
- Prepared Schema for advanced Sync Rules

**Comment by prashant-elastic** [comment]
**Updates [10th July - 14th July]**

- Tested GitHub connector with GraphQL implementation
- Tested E2E tests as per GraphQL implementation
- Started with the testing of Github advance sync rules. Currently, few points related to sync rules are in discussion over a [slack thread](https://crestdatasystems.slack.com/archives/C02AV5C8ZDE/p1689760548505479). Hence, once we have a clarity and we conclude approach for the sync rules, will continue testing with updated implementation approach

**Comment by abhishekjoshi-crest** [comment]
**Updates [13th July - 20th July]**
- Address review comments on GraphQL implementation #1289 
- Raised E2E PR with GraphQL Implementation. Waiting for Tim review #1296 
- Completed Advanced Sync Rules Implementation and waiting on few point to be discussed will update accordingly

**Comment by abhishekjoshi-crest** [comment]
Updates [21th July - 27th July]
- E2E PR with GraphQL Implementation was merged #1296 
- Raised PR for Advanced Sync Rules and its approved by sean #1324 
- Working on indexing markdown files only

**Comment by prashant-elastic** [comment]
Updates [21th July - 27th July]

- Testing of Github sync rules has been completed and PR has been raised

**Comment by danajuratoni** [comment]
thanks @prashant-elastic! will be picked up for review early next week! cc: @timgrein

**Comment by prashanttest-elastic** [comment]
@danajuratoni Recorded a demo video for Github connector and it is available in Google Drive:
[GitHub connector demo](https://drive.google.com/file/d/1dX5RRS_u7NixssUlSyNYRsESXCRMaqPu/view?usp=sharing)

**Comment by prashant-elastic** [comment]
@danajuratoni Recorded a demo video for Github connector and it is available in Google Drive:
[GitHub connector demo](https://drive.google.com/file/d/1dX5RRS_u7NixssUlSyNYRsESXCRMaqPu/view?usp=sharing)

**Comment by timgrein** [comment]
LGTM, signed it 👍

**Comment by danajuratoni** [comment]
Awesome! Closing as done! 🙌
